test/metabase/api/database_test.clj

```diff
@@ -511,6 +517,42 @@
                                                {:settings {:database-enable-actions true}})
                          [:settings :database-enable-actions]))))))
 
+(deftest update-database-enable-actions-open-connection-test
+  (testing "Updating a database's `database-enable-actions` setting shouldn't close existing connections (metabase#27877)"
+    (mt/test-drivers (filter #(isa? driver/hierarchy % :sql-jdbc) (mt/normal-drivers-with-feature :actions))
+      (let [;; 1. create a database and sync
+            database-name      (name (gensym))
+            empty-dbdef        {:database-name database-name}
+            _                  (tx/create-db! :postgres empty-dbdef)
+            connection-details (tx/dbdef->connection-details driver/*driver* :db empty-dbdef)
+            db                 (first (t2/insert-returning-instances! :model/Database {:name    database-name
+                                                                                       :engine  (u/qualified-name driver/*driver*)
+                                                                                       :details connection-details}))
+            _                  (sync/sync-database! db)]
+        (let [;; 2. start a long running process on another thread that uses a connection
+              connections-stay-open? (future
+                                       (sql-jdbc.execute/do-with-connection-with-options
+                                        driver/*driver*
+                                        db
+                                        nil
+                                        (fn [^Connection conn]
+                                          ;; sleep long enough to make sure the PUT request below finishes processing,
+                                          ;; including any async operations that it might trigger
+                                          (Thread/sleep 1000)
```

Comment:
If you have a smarter way to do this, I'm all ears.

Reply:
Heh, that's one interesting problem, given the 'any async operations' qualifier. :) I wonder if it makes sense to use with-redefs to capture async stuff?.. Or do we actually have something for like gathering/capturing async stuff for tests?

Reply:
Oh, how do you like idea like that: redefine `driver/notify-database-updated` to check it's not called?

Reply:
> redefine driver/notify-database-updated to check it's not called
Would that solve the issue? `driver/notify-database-updated` might not be called at the time of the check for two reasons:
1. `driver/notify-database-updated` has not been called and won't be called in the future
2. `driver/notify-database-updated` will be called in the future by code that executes in the future

If we check whether `driver/notify-database-updated` is called, we need to make sure that we check it after some time so 2 is not a possibility. Which means we need to do a `Thread/sleep` anyway.

Reply:
At least the way this test is implemented, it has no mention of `driver/notify-database-updated`, which is more of an implementation detail than a property we actually want to test here which is that the connection never gets closed.

Reply:
Ah, right, absolutely! Well, the only improvement is grabbing onto a start of various async stuff, and it feels a bit out of scope for this PR. :)

Reply:
Yeah. Do you have a quick explanation on what that could look like?

Reply:
Well, if we have just a single starting point for async stuff, `with-redefs` could handle this: collect all the started processes in an atom and return it or wait on them (up to implementation). Could be a nice macro like `(gather-async-processes ...)`, of course.

The bigger problem is if there are a lot of various ways to run them (like `future`, and separate thread pools, and `core.async`), but I think that requires clean up/unification then. :)